packages/blade-old/MIGRATION.md

```diff
@@ -0,0 +1,195 @@
+- [From version 1.4.x to 2.x.x](#from-version-14x-to-2xx)
+
+
+
+# From version 1.4.x to 2.x.x
+
+## Migration guide for new blade-old icons
+
+### Migrating icons to new API
+
+
+```diff
+- import Icon from "@razorpay/blade-old";
++ import { AlertCircle } from "@razorpay/blade-old/icons";
+
+- <Icon name="alert-circle" fill="shade.950" size="medium" />
++ <AlertCircle fill="shade.950" size="medium" />
+```
+
+To make this process easier I would suggest you use vscode search panel to search all the instances of `name="` which will show you all the places you used icons, and replace them with their respective icon.
+
+
+## Changes in \<Icon />
+
+Icon component is now a generic wrapper to create new icons. 
+To add new icons to blade's icon library you can use the <Icon /> component like so: 
+
+```jsx
+// User defined custom icons
+export const MyCustomIcon = (props) => {
+  return (
+    <Icon viewBox="0 0 24 24" {...props}>
+      <path d="" />
+    </Icon>
+  );
+};
+
+// usage
+<Button icon={MyCustomIcon}>
+```
+
+## Migrating existing blade-old components
+
+Few of the internal components were using the old \<Icon /> component, thus we also had to change the APIs for them.
+
+```sh
+# migrate-icons [directory_path_glob] [...jscodeshift_options]
+migrate-icons src/App/** parser=jsx
+```
+
+### Automigration
+
+We've written a codemod to automate the process of replacing the `icon` prop with imported icons. 
+
+Simply install the blade-old package and run
+
+```sh
+// migrate-icons [directory_path_glob] [...jscodeshift_options]
```

Comment:
```suggestion
// migrate-icons [directory_path_glob] [...jscodeshift_options]
cd packages/blade-old
```

Reply:
Didn't get this 🤔 There won't be any `cd packages/blade-old`

The usage flow will be like this lets say:

1. User scaffolds a new universe react project
2. They used the older API for the Icons
3. They will upgrade blade-old
4. Then they will run the bin command directly `migrate-icons [path]` 
5. Icons will be migrated